src/csvcubed/utils/qb/validation/observations.py

```diff
@@ -35,35 +49,67 @@ def validate_observations(cube: Cube) -> List[ValidationError]:
 
     num_obs_val_columns = len(observed_value_columns)
-    # strtobool will work for "True" and "true" inputs. And similarly, for "False" and "false". 
-    is_pivoted_multi_measure = strtobool(os.environ.get("PIVOTED_MULTI_MEASURE", 'False'))
-            
+    # strtobool will work for "True" and "true" inputs. And similarly, for "False" and "false".
+    is_pivoted_multi_measure = strtobool(
+        os.environ.get("PIVOTED_MULTI_MEASURE", "False")
+    )
+
     if num_obs_val_columns == 0:
         errors.append(NoObservedValuesColumnDefinedError())
     elif num_obs_val_columns > 1 and not is_pivoted_multi_measure:
         errors.append(MoreThanOneObservationsColumnError(num_obs_val_columns))
     else:
-        pivoted_obs_val_columns = [
+        obs_val_columns_with_measure = [
             c
             for c in observed_value_columns
-            if c.structural_definition.is_pivoted_shape_observation
+            if c.structural_definition.measure is not None
         ]
-        standard_shape_obs_val_columns = [
-            c
-            for c in observed_value_columns
-            if not c.structural_definition.is_pivoted_shape_observation
+        obs_val_columns_without_measure = [
+            c for c in observed_value_columns if c.structural_definition.measure is None
         ]
-        
-        if len(standard_shape_obs_val_columns) == 1:
-            obs_val_column = standard_shape_obs_val_columns[0]
-            errors += _validate_observation_value(obs_val_column, multi_units_columns)
-            errors += _validate_standard_shape_cube(cube)
-        elif len(pivoted_obs_val_columns) == 1:
-            obs_val_column = pivoted_obs_val_columns[0]
-            errors += _validate_observation_value(obs_val_column, multi_units_columns)
-            errors += _validate_pivoted_shape_cube(cube)
-
-        errors += _validate_missing_observation_values(cube, observed_value_columns[0])
+
+        errors += _get_shape_related_errors(
+            cube,
+            obs_val_columns_with_measure,
+            obs_val_columns_without_measure,
+            multi_units_columns,
+            observed_value_columns,
+        )
+
+    return errors
+
+
+def _get_shape_related_errors(
+    cube: Cube,
+    obs_val_columns_with_measure: list[QbColumn[QbObservationValue]],
+    obs_val_columns_without_measure: list[QbColumn[QbObservationValue]],
+    multi_units_columns: List[QbColumn[QbMultiUnits]],
+    observed_value_columns: list[QbColumn[QbObservationValue]],
+) -> List[ValidationError]:
+    errors: List[ValidationError] = []
+
+    if any(obs_val_columns_with_measure) and any(obs_val_columns_without_measure):
+        errors += _get_error_when_cube_contains_obs_val_cols_with_and_without_measures(
+            cube
+        )
+
+    elif len(obs_val_columns_without_measure) > 1:
+        errors += _get_error_when_cube_only_contains_obs_val_cols_without_measures(
```

Comment:
Can you rename this function to make it clear that it only applies when we are in the pivoted shape?

Reply:
Renamed